librz/core/cil.c

```diff
@@ -183,21 +180,9 @@ RZ_IPI void rz_core_analysis_esil_init_mem(RzCore *core, const char *name, ut64
 			break;
 		}
 	}
-	// SP
-	const char *sp = rz_reg_get_name(core->dbg->reg, RZ_REG_NAME_SP);
-	if (sp) {
-		rz_debug_reg_set(core->dbg, sp, addr + (size / 2));
-	}
-	// BP
-	const char *bp = rz_reg_get_name(core->dbg->reg, RZ_REG_NAME_BP);
-	if (bp) {
-		rz_debug_reg_set(core->dbg, bp, addr + (size / 2));
-	}
-	// PC
-	const char *pc = rz_reg_get_name(core->dbg->reg, RZ_REG_NAME_PC);
-	if (pc) {
-		rz_debug_reg_set(core->dbg, pc, current_offset);
-	}
+	rz_reg_set_value_by_role(core->analysis->reg, RZ_REG_NAME_SP, addr + (size / 2));
```

Comment:
Could you please add a comment on why it is `size / 2`?

Reply:
I can only assume the purpose. I think it is to have room to both sides.